src/utils/fetchUtils.ts

```diff
@@ -224,3 +231,62 @@ export async function fetchLanguages(): Promise<ILanguage[]> {
   });
   return languages;
 }
+
+function sortByHourAndMin(a: IEvent, b: IEvent) {
+  const aStart = a.dateStart;
+  const bStart = b.dateStart;
+  if (aStart.getHours() - bStart.getHours() === 0) {
+    return aStart.getMinutes() - bStart.getMinutes();
+  }
+  return aStart.getHours() - bStart.getHours();
+}
+
+// Since some dates in API have actual hours and others are always at 00:00
+// (meaning entire days presumably), we compare by start and end of day
+function isIncludedInDateRange(start: string, end: string) {
+  const rangeStartDate = startOfDay(new Date(start));
+  const rangeEndDate = endOfDay(new Date(end));
+  return (item: IEvent) => {
+    const { dateStart, dateEnd } = item;
+    const zonedStart = startOfDay(zonedTimeToUtc(dateStart, TZ));
+    const zonedEnd = endOfDay(zonedTimeToUtc(dateEnd, TZ));
+    return zonedStart <= rangeStartDate && zonedEnd >= rangeEndDate;
+  };
+}
+
+export async function fetchEvents(
+  userGroupId: number,
+  lang?: string,
+  dateStart?: string,
+  dateEnd?: string,
+): Promise<IEvent[]> {
+  const url = buildEventsUrl(userGroupId, lang, dateStart, dateEnd);
+  logApp(`sending fetch request to: ${url}`);
+
+  // TODO: refactor this?
+  const response = await fetch(url);
+  logApp(`received fetching response from: ${url}`);
+  if (!response.ok) {
+    throw new Error("Malformed request");
+  }
+
+  let events: IEvent[] = [];
+  const eventsJson = await response.json();
+  eventsJson.forEach((data: any) => {
+    try {
+      const parsedEvents = parseEvent(data);
+      events.push(...parsedEvents);
+    } catch (error) {
+      // Discard item
+      logWarn("Failed to parse Event: ", data);
+      logWarn("Validation error: ", error);
+    }
+  });
+  if (dateStart && dateEnd) {
+    const filterFn = isIncludedInDateRange(dateStart, dateEnd);
+    const filteredEvents = events.filter(filterFn);
+    events = filteredEvents;
+  }
+  const sortedEvents = events.sort(sortByHourAndMin);
+  return sortedEvents;
```

Comment:
Could combine these two lines into one

Reply:
For the return variable I kind of like having a more explicit name, partly because there might be more sorting/filtering/whatever added later, but for the previous one (`filteredEvents`) I agree!